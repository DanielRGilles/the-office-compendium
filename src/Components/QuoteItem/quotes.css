.quote-item > img {
    width: 75%;
    padding: 15px;
}
.quote-item {
    border: 2px ridge black;
    border-radius: 4px;
    display: flex;
    flex-direction: column;
    align-items: center;
    background-color: rgb(248, 241, 228);
    margin-bottom: 3px;
}
ul {
    margin: 0;
    padding: 5px;
}
.quote-text {
    width: 75%;
    font-weight: 400;
    background-color: rgb(184, 216, 243);
    padding: 5px;
    border-radius: 5px;
    box-shadow: 0px 1px 1px rgb(128, 126, 126);
    margin-bottom: 3px;
}
