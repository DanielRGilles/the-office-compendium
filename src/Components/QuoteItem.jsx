import React from 'react';
import './quotes.css';

export default function QuoteItem({quote}) {
    return (
        <li  className='quote-item'>
            <img 
            src={`Assets/${quote.character.firstname}.jpg`}  alt={quote.character.firstname}/>
            <div className='quote-text' key={quote._id}>{quote.character.firstname} {quote.character.lastname} 
                        : {quote.content}</div>
        </li>
    )
}
