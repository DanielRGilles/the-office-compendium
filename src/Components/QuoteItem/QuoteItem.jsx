import React from 'react';
import './quotes.css'

export default function QuoteItem({quote}) {
    return (
        <li  className='quote-item'  key={quote._id}>
            <img 
            src={`Assets/${quote.character.firstname}.jpg`}  alt={quote.character.firstname}/>
            <div className='quote-text'>{quote.character.firstname} {quote.character.lastname} 
                        : {quote.content}</div>
        </li>
    )
}
