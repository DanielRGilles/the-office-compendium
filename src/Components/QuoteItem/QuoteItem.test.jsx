import { render } from '@testing-library/react'
import QuoteItem from './QuoteItem'

it('Should fetch then render a list of office quotes', async () => {
  const quote = {
    "_id": "5e966a526a66e65486e244b5",
    "content": "Shut up about the sun. SHUT UP ABOUT THE SUN!",
    "character": {
        "_id": "5e93b56d3af44260882e33be",
        "firstname": "Gabe",
        "lastname": "Lewis",
        "__v": 0
    },
    "__v": 0
};

const { container } = render(<QuoteItem quote={quote} />)
  expect(container).toMatchSnapshot()
})
