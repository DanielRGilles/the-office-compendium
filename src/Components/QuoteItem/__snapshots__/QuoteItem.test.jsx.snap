// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should fetch then render a list of office quotes 1`] = `
<div>
  <li
    class="quote-item"
  >
    <img
      alt="Gabe"
      src="Assets/Gabe.jpg"
    />
    <div
      class="quote-text"
    >
      Gabe
       
      Lewis
      : 
      Shut up about the sun. SHUT UP ABOUT THE SUN!
    </div>
  </li>
</div>
`;
